static/js/subs.js

```diff
@@ -689,9 +694,23 @@ exports.new_stream_clicked = function () {
     stream_create.new_stream_clicked(stream);
 };
 
+exports.display_unsubscribe_prompt = function (sub) {
+    var unsubscribe_stream_modal = templates.render('unsubscribe-stream-modal', {stream_name: sub.name});
+    $(".subscription_settings").append(unsubscribe_stream_modal);
+    overlays.open_modal("unsubscribe_stream_modal");
+    $("#unsubscribe_stream_modal").on("click", "#do_unsubscribe_stream_button", function () {
+        $("#unsubscribe_stream_modal").modal("hide");
+        ajaxUnsubscribe($("#do_unsubscribe_stream_button").data("stream"), true);
```

Comment:
Here, maybe you want to add listener which delete the unsubscription warning modal from DOM on any option clicked `Ok` or `Cancel`, otherwise we will have two same modals in DOM, resulted in error. You can check out implementation of stream deletion modal for reference.